modules/oe_translation_cdt/src/Controller/OperationController.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\oe_translation_cdt\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\oe_translation_cdt\Api\CdtApiWrapperInterface;
+use Drupal\oe_translation_cdt\ContentFormatter\ContentFormatterInterface;
+use Drupal\oe_translation_cdt\Mapper\LanguageCodeMapper;
+use Drupal\oe_translation_cdt\TranslationRequestCdtInterface;
+use Drupal\oe_translation_cdt\TranslationRequestUpdaterInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Returns responses for OpenEuropa Translation CDT routes.
+ */
+final class OperationController extends ControllerBase {
+
+  /**
+   * Constructs the controller.
+   *
+   * @param \Drupal\oe_translation_cdt\Api\CdtApiWrapperInterface $apiWrapper
+   *   The CDT API wrapper.
+   * @param \Drupal\oe_translation_cdt\TranslationRequestUpdaterInterface $updater
+   *   The translation request updater.
+   * @param \Drupal\oe_translation_cdt\ContentFormatter\ContentFormatterInterface $xmlFormatter
+   *   The XML formatter.
+   */
+  public function __construct(
+    private readonly CdtApiWrapperInterface $apiWrapper,
+    private readonly TranslationRequestUpdaterInterface $updater,
+    private readonly ContentFormatterInterface $xmlFormatter
+  ) {}
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): self {
+    return new self(
+      $container->get('oe_translation_cdt.api_wrapper'),
+      $container->get('oe_translation_cdt.translation_request_updater'),
+      $container->get('oe_translation_cdt.xml_formatter')
+    );
+  }
+
+  /**
+   * Refreshes the request status.
+   */
```

Comment:
In the context of drupal we must document all parameters (this is valid everywhere), unless the convention changed and im unaware

Reply:
How on Earth it passed my pipelines... I am fixing it.